### PR TITLE
Ag pet survey

### PR DESCRIPTION
**Pull request by adamrp**
Adds pet survey, associated menu items, and various supporting functions, database tables, and procedures. Also changes validation on text boxes that use validateText to be more restrictive.

### CODE REVIEW

File: python_code/data_access/ag_data_access.py [hunk 1]
```diff
@@ -281,6 +285,36 @@ def deleteSample(self, barcode, ag_login_id):
...
+        for row in results:
+            participants.append(row[0])
+
+        return participants
+
+    def getParticipantExceptions(self, ag_login_id):
+        con = self.getMetadataDatabaseConnection()
+        results = con.cursor()
+        con.cursor().callproc('ag_get_participant_exceptions', [ag_login_id, results])
+        exceptions = []
```
Comment (ElDeveloper) [inline python_code/data_access/ag_data_access.py]: These three lines could be reduced using list comprehension but I guess it's fine as it is?

File: www/americangut/animal_survey.psp [hunk 2]
```diff
@@ -189,37 +227,45 @@
...
+                    <table id="survey" width="100%">
+                        <colgroup>
+                           <col span="1" style="width: 40%;">
+                           <col span="1" style="width: 60%;">
+                        </colgroup>
+                        <tr>
+                            <td>Toilet water access</td>
+                            <td>
+                                <div id="toilet">
+                                    <input tabindex="" type="radio" id="toilet_regular" name="toilet" value="regular" /><label for="toilet_regular">Regular</label>
```
Comment (ElDeveloper) [inline www/americangut/animal_survey.psp]: All tab indices are empty, perhaps worth indexing those?

**Comment by douginator2000** [comment]
Is this ready to be pulled? Also curious why it's not auto-mergeable.

**Comment by ElDeveloper** [comment]
If the date picker has been removed, I think it's good to go. Else I don't really know.

@adamrp?

**Comment by adamrp** [comment]
Yeah, I think it's ready to go.  It's not auto-mergeable proabably because of conflicts with menu.psp, but I can't be sure.  We'll know when we try to merge...  If you'd like I can sit down with you while you merge the changes.

**Comment by douginator2000** [comment]
Shouldn't be any conflicts - I don't want to do a merge on the live system
to test out conflicts with another codebase. Suggest pulling the menu
changes on your branch, resolving, testing, and pushing an updated copy
into the pull request so that we're sure the code going live is working.

On Tue, Jul 2, 2013 at 11:45 AM, adamrp notifications@github.com wrote:

> Yeah, I think it's ready to go. It's not auto-mergeable proabably because
> of conflicts with menu.psp, but I can't be sure. We'll know when we try to
> merge... If you'd like I can sit down with you while you merge the changes.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/qiime/qiime_web_app/pull/449#issuecomment-20362703
> .

## 

Doug Wendel
Professional Research Associate
University of Colorado at Boulder
Department of Chemistry and Biochemistry, Knight Lab
303-641-5442
wendel@colorado.edu

**Comment by ElDeveloper** [comment]
:+1: Yes, that's exactly the right way to do it.

**Comment by adamrp** [comment]
Resolved conflicts, I'll leave it to someone else to merge it into master.

**Comment by mikewise** [comment]
Just re-pinging to see if we can get this merged, now that it looks like all the errors have been resolved. We've been getting plenty of participant emails eager to start using the pet survey to log samples.

**Comment by douginator2000** [comment]
Hey Mike, yes I'd like to pull this, though I do want to wait until the
development database server is running again so we can deploy it once in a
test environment before making it live.

On Thu, Jul 11, 2013 at 11:37 AM, mikewise notifications@github.com wrote:

> Just re-pinging to see if we can get this merged, now that it looks like
> all the errors have been resolved. We've been getting plenty of participant
> emails eager to start using the pet survey to log samples.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/qiime/qiime_web_app/pull/449#issuecomment-20828246
> .

## 

Doug Wendel
Professional Research Associate
University of Colorado at Boulder
Department of Chemistry and Biochemistry, Knight Lab
303-641-5442
wendel@colorado.edu

**Comment by mikewise** [comment]
No Problem, I didn't know that the dev server was down.

**Comment by adamrp** [comment]
This was deployed on webdev for a while (back when I did my code review on it and asked everyone to test it out).  I don't think anyone found any problems with it, but if we want to take another even closer look once webdev comes back up, that's fine.  There's quite a few changes, so if you want to actually sit down and go through it, I'd be happy to do so (whether sooner or later).

**Comment by ElDeveloper** [comment]
I remember this PR being good to go. But we might want to give it another pass.

Thanks!

Yoshiki.

**Pull request merged**